app/models/user.rb

```diff
@@ -540,4 +546,33 @@ def update_pseud_name
    def log_change_if_login_was_edited
      create_log_item( options = {:action => ArchiveConfig.ACTION_RENAME, :note => "Old Username: #{login_was}; New Username: #{login}"}) if login_changed?
    end
+
+  def self.perform(type)
+    case type
+    when 'update_last_login'
+      self.update_last_login
+    end
+  end
+
+  # called from rake
+  def self.update_last_login
+    User.transaction do
+      list = REDIS_GENERAL.smembers("last_login_list").each
+      list.each do |username|
```

Comment:
would change this to userid throughout for clarity and make it user = User.find(userid) instead but otherwise I don't think the code needs to be changed in any way